pkg/syncer/metadatasyncer.go

```diff
@@ -984,16 +991,21 @@ func ReloadConfiguration(metadataSyncer *metadataSyncInformer, reconnectToVCFrom
 				}
 				// Remove Old Volume Managers which are replaced with new VC IP or FQDN
 				for vcHost := range metadataSyncer.configInfo.Cfg.VirtualCenter {
-					var retainVolumeManager bool
+					var retainOldVc bool
 					for _, newconfig := range newVcenterConfigs {
 						if newconfig.Host == vcHost {
-							retainVolumeManager = true
+							retainOldVc = true
 							break
 						}
 					}
-					if !retainVolumeManager {
+					// If VC host has changed to FQDN from VC IP or vice versa,
+					// update maps volumeOperationsLock, cnsCreationMap and cnsDeletionMap.
+					if !retainOldVc {
 						log.Infof("deleting volume manager for vCenter: %q", vcHost)
 						delete(metadataSyncer.volumeManagers, vcHost)
+						delete(volumeOperationsLock, vcHost)
```

Comment:
Can we have a lock between full sync and reloadconfig to protect cnsCreationMap and cnsDeletionMap.
only one should be operating on cnsCreationMap and cnsDeletionMap at a time.

If fullsync is running, lets hold ReloadConfiguration and if ReloadConfiguration is going on, lets hold fullsync

Reply:
Added a lock to prevent race between reload configuration, fullSync and metadatasyncer